keras_cv/models/object_detection/faster_rcnn.py

```diff
@@ -0,0 +1,326 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+from keras_cv.bounding_box.converters import _decode
+from keras_cv.layers.object_detection.anchor_generator import AnchorGenerator
+from keras_cv.layers.object_detection.roi_align import _ROIAligner
+from keras_cv.layers.object_detection.roi_generator import ROIGenerator
+from keras_cv.layers.object_detection.roi_sampler import _ROISampler
+from keras_cv.layers.object_detection.rpn_label_encoder import _RpnLabelEncoder
+from keras_cv.ops.box_matcher import ArgmaxBoxMatcher
+
+
+def _resnet50_backbone():
+    inputs = tf.keras.layers.Input(shape=(None, None, 3))
+    x = inputs
+
+    backbone = tf.keras.applications.ResNet50(include_top=False, input_tensor=x)
+
+    c2_output, c3_output, c4_output, c5_output = [
+        backbone.get_layer(layer_name).output
+        for layer_name in [
+            "conv2_block3_out",
+            "conv3_block4_out",
+            "conv4_block6_out",
+            "conv5_block3_out",
+        ]
+    ]
+    return tf.keras.Model(
+        inputs=inputs, outputs=[c2_output, c3_output, c4_output, c5_output]
+    )
+
+
+class FeaturePyramid(tf.keras.layers.Layer):
```

Comment:
lets split FPN into another file even if its just internal.  Much better for readability

Reply:
I left them as a single file because we should be able to remove them soon -- from Scott's PR.

Reply:
This FPN is bit different from the one in retinanet, Is that expected?

Reply:
Yes, it is expected